py3status/modules/sysdata.py

```diff
@@ -195,16 +195,20 @@ def _get_stat(self):
                 'idle': int(fields[4]),
             }
 
-    def _calc_mem_info(self, unit='GiB', memi=dict, keys=list):
+    def _calc_mem_info(self, unit='GiB', memi=dict, keys=list, mem=bool):
         """
         Parse /proc/meminfo, grab the memory capacity and used size
         then return; Memory size 'total_mem', Used_mem, percentage
         of used memory, and units of mem (KiB, MiB, GiB).
         """
         total_mem_kib = memi[keys[0]]
-        total_used_mem_kib = memi[keys[0]] - memi[keys[1]]
-        cached_mem_kib = memi[keys[3]] + memi[keys[5]] - memi[keys[4]]
-        used_mem_kib = total_used_mem_kib - (memi[keys[2]] + cached_mem_kib)
+        if mem:
+            total_used_mem_kib = memi[keys[0]] - memi[keys[1]]
+            cached_mem_kib = memi[keys[3]] + memi[keys[5]] - memi[keys[4]]
+            used_mem_kib = total_used_mem_kib - (memi[keys[2]] + cached_mem_kib)
+        else:
+            used_mem_kib = total_mem_kib - memi[keys[1]]
+
```

Comment:
this code is terrible.

Can we just make two functions one for swap and one for memory and kill all the `memi[keys[2]]` junk memi['Buffers:']` is vaguely readable what we have now it crazy

Reply:
Minimal diff is the game here. We can do this later. Like you said, focus on important PRs.

I want to add other placeholders too. Print `612 MiB/8 GiB` instead of `0.53/8 GiB`.
* `{mem_used}        memory used, eg 7.17`
* `{mem_used_unit}   memory used unit, eg GiB`
* `{mem_total}       memory total, eg 7.72`
* `{mem_total_unit}  memory total unit, eg GiB`

Is that OK? I just wanted to fix this minor `swap` issue and get out of here.

Reply:
> Minimal diff is the game here.  

Readable code is the game :)

Reply:
Read them.